Orange/statistics/util.py

```diff
@@ -607,59 +607,35 @@ def std(x, axis=None, ddof=0):
     return np.sqrt(var(x, axis=axis, ddof=ddof))
 
 
-# to speed-up FDR, calculate ahead sum([1/i for i in range(1, m+1)]),
-# for m in [1,100000].
-# For higher values of m use an approximation, with error less or equal to
-# 4.99999157277e-006. (sum([1/i for i in range(1, m+1)])  ~ log(m) + 0.5772...,
-#  0.5572 is an Euler-Mascheroni constant)
-c = [1.0]
-for m in range(2, 100000):
-    c.append(c[-1] + 1.0/m)
-
-
-def FDR(p_values, dependent=False, m=None, ordered=False):
+def FDR(p_values: Iterable, dependent=False, m=None, ordered=False) -> Iterable:
     """ `False Discovery Rate <http://en.wikipedia.org/wiki/False_discovery_rate>`_
         correction on a list of p-values.
 
-    :param p_values: a list of p-values.
+    :param p_values: list or np.ndarray of p-values.
     :param dependent: use correction for dependent hypotheses (default False).
     :param m: number of hypotheses tested (default ``len(p_values)``).
     :param ordered: prevent sorting of p-values if they are already sorted
         (default False).
+    :return: list or a np.ndarray - depends on the input
     """
-    def is_sorted(l):
-        return all(l[i] <= l[i + 1] for i in range(len(l) - 1))
-
-    if not ordered:
-        ordered = is_sorted(p_values)
+    if p_values is None or len(p_values) == 0 or \
+            (m is not None and m <= 0):
+        return None
```

Comment:
Cover this line in tests to get approval by codecov (and to make sure we don't forget these cases at any future refactoring).